src/main/java/org/elasticsearch/index/query/RangeQueryParser.java

```diff
@@ -123,11 +123,6 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         FieldMapper mapper = parseContext.fieldMapper(fieldName);
         if (mapper != null) {
             if (mapper instanceof DateFieldMapper) {
-                if ((from instanceof Number || to instanceof Number) && timeZone != null) {
```

Comment:
Shouldn't we still have this? It doesn't really make sense to set seconds or milliseconds since epoch in anything but UTC?